subgraph/schema.graphql

```diff
@@ -0,0 +1,41 @@
+type Asset @entity(immutable: true) {
+  id: Bytes!
+  address: Bytes!
+  decimals: Int!
+  label: String!
+}
+
+type Offer @entity {
+  id: Bytes!
+  domainName: String!
+  sourceAsset: Asset!
+  destAsset: Asset!
+  offerAddress: Bytes!
+  domainOwner: Bytes!
+  depositHistory: [BigInt!]!
+  totalDeposits: BigInt!
+  closeAmount: BigInt!
+  commissionRate: BigInt!
+  lockWithdrawAfter: BigInt!
+  events: [Event!]! @derivedFrom(field: "offer")
```

Comment:
same here - not sure if we need events to be part of Offer, instead of just storing them separately, and just store offer id in the events, rather than having offer linked to events inside offer

Reply:
`events` field is not stored explicitly in `Offer` entity in db. `Event` entity has `offer` field, and when the same offer is added to multiple `Event` entities, the `Offer` entity will calculate its `events` field automatically. This is what `@derivedFrom` keyword does.